test-monitor.cpp

```diff
@@ -1,8 +1,119 @@
+}
+
+// Test Case: SpO2 Just Below Lower Limit
+TEST(AreVitalsNormalTest, Spo2JustBelowLowerLimit) {
+    EXPECT_FALSE(areVitalsNormal(98.6, 75, SPO2_LOWER_LIMIT));
+}
+
+// Test Case: SpO2 Just Above Lower Limit
+TEST(AreVitalsNormalTest, Spo2JustAboveLowerLimit) {
+    EXPECT_TRUE(areVitalsNormal(98.6, 75, SPO2_LOWER_LIMIT + 0.1));
+}
+
+// Test Case: SpO2 Just Below Upper Limit
+TEST(AreVitalsNormalTest, Spo2JustBelowUpperLimit) {
+    EXPECT_TRUE(areVitalsNormal(98.6, 75, SPO2_UPPER_LIMIT - 0.1));
+}
+
+// Test Case: SpO2 Just Above Upper Limit
+TEST(AreVitalsNormalTest, Spo2JustAboveUpperLimit) {
+    EXPECT_FALSE(areVitalsNormal(98.6, 75, SPO2_UPPER_LIMIT));
+}
+
+// Test Case: All Vitals Within Normal Range
+TEST(AreVitalsNormalTest, AllVitalsWithinNormalRange) {
+    EXPECT_TRUE(areVitalsNormal(98.6, 75, 95));
+}
+
+// Test Case: All Vitals Just Inside Lower Limits
+TEST(AreVitalsNormalTest, AllVitalsJustInsideLowerLimits) {
+    EXPECT_TRUE(areVitalsNormal(TEMPERATURE_LOWER_LIMIT + 0.1, PULSE_RATE_LOWER_LIMIT + 0.1,
+                                SPO2_LOWER_LIMIT + 0.1));
+}
+
+// Test Case: All Vitals Just Inside Upper Limits
+TEST(AreVitalsNormalTest, AllVitalsJustInsideUpperLimits) {
+    EXPECT_TRUE(areVitalsNormal(TEMPERATURE_UPPER_LIMIT - 0.1, PULSE_RATE_UPPER_LIMIT - 0.1,
+                                SPO2_UPPER_LIMIT - 0.1));
+}
+
+// Test Case: All Vitals Out of Range (Temperature out of range)
+TEST(AreVitalsNormalTest, TemperatureOutOfRange) {
+    EXPECT_FALSE(areVitalsNormal(TEMPERATURE_LOWER_LIMIT - 0.1, 75, 95));
+}
+
+// Test Case: All Vitals Out of Range (Pulse Rate out of range)
+TEST(AreVitalsNormalTest, PulseRateOutOfRange) {
+    EXPECT_FALSE(areVitalsNormal(98.6, PULSE_RATE_UPPER_LIMIT + 0.1, 95));
+}
+
+// Test Case: All Vitals Out of Range (SpO2 out of range)
+TEST(AreVitalsNormalTest, Spo2OutOfRange) {
+    EXPECT_FALSE(areVitalsNormal(98.6, 75, SPO2_LOWER_LIMIT - 0.1));
+}
+
+// Test Case: All Vitals Borderline (Temperature at upper limit, Pulse Rate at upper limit,
+// SpO2 at lower limit)
+TEST(AreVitalsNormalTest, VitalsBorderline) {
+    EXPECT_FALSE(areVitalsNormal(TEMPERATURE_UPPER_LIMIT, PULSE_RATE_UPPER_LIMIT,
+                                SPO2_LOWER_LIMIT));
+}
+
+// Test Case: Only Temperature Out of Range (Temperature above upper limit, Pulse Rate
+// and SpO2 within range)
+TEST(AreVitalsNormalTest, OnlyTemperatureOutOfRange) {
+    EXPECT_FALSE(areVitalsNormal(TEMPERATURE_UPPER_LIMIT + 0.1, 75, 95));
+}
+
+// Test Case: Only Pulse Rate Out of Range (Pulse Rate above upper limit, Temperature
+// and SpO2 within range)
+TEST(AreVitalsNormalTest, OnlyPulseRateOutOfRange) {
+    EXPECT_FALSE(areVitalsNormal(98.6, PULSE_RATE_UPPER_LIMIT + 0.1, 95));
+}
 
-TEST(Monitor, NotOkWhenAnyVitalIsOffRange) {
-  ASSERT_FALSE(vitalsOk(99, 102, 70));
-  ASSERT_TRUE(vitalsOk(98.1, 70, 98));
+// Test Case: Only SpO2 Out of Range (SpO2 above upper limit, Temperature and

```

Comment:
Oxygen saturation cannot be more than 100% practically.
Do you think the test is still needed?